File: Arrays & Hashing/two sums.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num2 = target - nums[i];
            if (map.containsKey(num2)) {
                return new int[]{map.get(num2), i};
            }
            map.put(nums[i], i);
        }
        return new int[]{}; 
    }
}

File: Arrays & Hashing/python/two_sums.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i in range(len(nums)): 
            num2 = target - nums[i]
            if num2 in nums[i + 1:]:
                return [i, nums.index(num2, i + 1)]